fe/fe-core/src/main/java/org/apache/doris/analysis/Analyzer.java

```diff
@@ -626,6 +628,11 @@ public TableRef resolveTableRef(TableRef tableRef) throws AnalysisException {
             }
         }
 
+        if (table.getType() == TableType.HUDI && table.getFullSchema().isEmpty()) {
+            // resolve hudi table's schema when table schema is empty from doris meta
```

Comment:
What if the schema changed in HMS? Can we keep the schema in Doris up to date?

Reply:
two case:
1. create hudi table with schema. when table schema has been changed in HMS，all hudi table's schema are not lost in HMS，query successful. if some field has been renamed or drop in HMS, query failed.
2. create a hudi table without schema. when table schema has been changed in HMS, it will resolve schema from hms when query.

Reply:
@Jibing-Li fix it， please check again

Reply:
LGTM. We can find a better cache mechanism later.